src/arrayMethodJoin.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinText = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof (separator) === 'undefined') {
+        joinText += this[i] + ',';
+      } else if (this[i] === null || typeof (this[i]) === 'undefined') {
+        joinText += ',';
```

Comment:
Maybe it is better to replace default value to the function's argument (separator = ',')

Reply:
agree, just add default separator